demos/viscoelastic/viscoelastic_loading.py

```diff
@@ -0,0 +1,517 @@
+
+# This example
+# -------------
+# We will simulate a viscoelastic loading and unloading problem based on a 2D version of the test case presented in Weerdesteijn et al. (2023).
+#
+# Let's get started! The first step is to import the `gadopt` module, which
+# provides access to Firedrake and associated functionality.
+
+from gadopt import *
+from gadopt.utility import step_func
+
+# Next we need to create a mesh of the mantle region we want to simulate. The Weerdesteijn test case is a 3D box 1500 km wide horizontally and 2891 km deep. To speed up things for this first demo, we consider a 2D domain, i.e. taking a vertical cross section through the 3D box.
+#
+# For starters let's use one of the default meshes provided by Firedrake, `RectangleMesh`. We have chosen 40 quadrilateral elements in the $x$ direction and 40 quadrilateral elements in the $y$ direction. It is worth emphasising that the setup has coarse grid resolution so that the demo is quick to run! For real simulations we can use fully unstructured meshes to accurately resolve important features in the model, for instance near coastlines or sharp discontinuities in mantle properties.  We can print out the grid resolution using `log`, a utility provided by G-ADOPT. (N.b. `log` is equivalent to python's `print` function, except that it simplifies outputs when running simulations in parallel.)
+#
+# On the mesh, we also denote that our geometry is Cartesian, i.e. gravity points
+# in the negative z-direction. This attribute is used by G-ADOPT specifically, not
+# Firedrake. By contrast, a non-Cartesian geometry is assumed to have gravity
+# pointing in the radially inward direction.
+#
+# Boundaries are automatically tagged by the built-in meshes supported by Firedrake. For the `RectangleMesh` being used here, tag 1 corresponds to the plane $x=0$; 2 to the $x=L$ plane; 3 to the $y=0$ plane; and 4 to the $y=D$ plane. For convenience, we can rename these to `left_id`, `right_id`, `bottom_id` and `top_id`.
+
+# +
+# Set up geometry:
+L = 1500e3  # length of the domain in m
+D = 2891e3  # Depth of domain in m
+nx = 40  # Number of horizontal cells
+nz = 40  # number of vertical cells
+
+# Let's print out the grid resolution in km
+log(f"Horizontal resolution {L/nx/1000} km")
+log(f"Vertical resolution {D/nz/1000} km")
+
+mesh = RectangleMesh(nx, nz, L, D, name="mesh", quadrilateral=True)
+mesh.cartesian = True
+
+# Shift top of mesh to z = 0.
```

Comment:
This needs a little more of a detailed justification. Why is this done?

Reply:
(ah I actually assumed z was already like this for setting up the depth profiles. I've just modified it in the intialise background profile now